openedx/frontend-app-course-authoring
# fix: tinymce render outside of editors

## Description

This bug was introduced via #1208. The cause of this bug was that `state` has two different definitions when trying to `TinyMceWidget` outside of the editors. One definition of `state` was an empty object coming from the editors redux store. The second definition of `state` had the required variables (`images`, `lmsEndpointUrl`, `studioEndpointUrl`, `learningContextId`, and  `isLibrary`). The former loaded last so component tried to render referencing an empty object. 

To fix the bug, `<TinyMceWidget />` was refactored to not use the redux store to fetch `images`, `lmsEndpointUrl`, `studioEndpointUrl`, `learningContextId` or `isLibrary`. Now `images`,`learningContextId` and `isLibrary` are passed down from a parent component, and `lmsEndpointUrl` and `studioEndpointUrl` referenced from the config value.

This change impacts the Course Author
Before:
<img width="1539" alt="Screenshot 2024-09-04 at 12 00 28 PM" src="https://github.com/user-attachments/assets/44fe5a8b-070d-494f-8de3-1b8fff0e478e">

After:
<img width="1539" alt="Screenshot 2024-09-04 at 12 14 31 PM" src="https://github.com/user-attachments/assets/69666c03-6b94-4e85-9a52-1f6b8cb32938">

## Testing instructions

Updates testing
1. Open the Updates page for a course
2. Add a new update
3. Confirm that you can add a new update
4. Edit an existing update
5. Confirm that you can edit existing update
6. Edit handouts confirm that you can edit handouts

Schedule and details testing
1. Open the schedules and details page
2. Confirm that the page loads
3. Scroll to the "Course overview"
4. Confirm that section loads as expected

Text editor testing
1. Open a text component for editing
2. Confirm that the editor opens and content loads as expected

Problem editor testing
1. Open a multi select problem for editing
2. Confirm that the editor opens and content loads as expected
3. Add feedback to one of the answer choices
4. Add a hint
5. Click "Save"
6. In the unit page view, complete the problem
7. Confirm that feedback and hint display as expected
8. Re-open the problem editor
9. Confirm that the editor opens and content loads as expected
